services/apps/test/apps/containers_test.clj

```diff
@@ -1,26 +1,89 @@
+(def ^:dynamic volumes-from-map nil)
 
-(def image-info-map (add-image-info {:name "discoenv/de-db" :tag "latest" :url "https://www.google.com"}))
+(defn- add-image-info-map []
+  (add-image-info {:name "discoenv/de-db" :tag "latest" :url "https://www.google.com"}))
 
-(def data-container-map
+(defn- add-data-container-map []
   (add-data-container {:name        "discoenv/foo"
                        :tag         "latest"
                        :url         "https://www.google.com"
                        :name_prefix "foo"
                        :read_only   true}))
 
+(defn- get-tool-map []
+  (first (select tools (where {:name "notreal"}))))
+
+(defn- add-settings-map [tool-id]
+  (add-settings {:name              "test"
+                 :cpu_shares        1024
+                 :memory_limit      2048
+                 :network_mode      "bridge"
+                 :working_directory "/work"
+                 :tools_id          tool-id}))
+
+(defn- add-devices-map [settings-id]
+  (add-device settings-id {:host_path "/dev/null" :container_path "/dev/yay"}))
+
+(defn- add-volume-map [settings-id]
+  (add-volume settings-id {:host_path "/tmp" :container_path "/foo"}))
+
+(defn- add-volumes-from-map [settings-id data-container-id]
+  (add-volumes-from settings-id data-container-id))
+
+(defn- add-test-data []
+  (let [image-info     (add-image-info-map)
+        data-container (add-data-container-map)
+        tool           (get-tool-map)
+        settings       (add-settings-map (:id tool))
+        device         (add-devices-map (:id settings))
+        volume         (add-volume-map (:id settings))
+        volumes-from   (add-volumes-from-map (:id settings) (:id data-container))]
+    (vector image-info data-container tool settings device volume volumes-from)))
+
+(defn- remove-container-image-references []
+  (sql/update tools
+              (set-fields {:container_images_id nil})
+              (where {:container_images_id (:id image-info-map)})))
+
+(defn- remove-test-data []
+  (delete container-volumes-from (where {:id (:id volumes-from-map)}))
+  (delete container-volumes (where {:id (:id volume-map)}))
+  (delete container-devices (where {:id (:id devices-map)}))
+  (delete container-settings (where {:id (:id settings-map)}))
+  (delete data-containers (where {:id (:id data-container-map)}))
+  (remove-container-image-references)
+  (delete container-images (where {:id (:id image-info-map)})))
+
+(defn- with-test-data [f]
+  (let [[image-info data-container tool settings device volume volumes-from] (add-test-data)]
+    (binding [image-info-map     image-info
+              data-container-map data-container
+              tool-map           tool
+              settings-map       settings
+              devices-map        device
+              volume-map         volume
+              volumes-from-map   volumes-from]
+      (f)
+      (remove-test-data))))
+
+(use-fixtures :each with-test-db run-integration-tests with-test-data)
+
```

Comment:
That would work. I've been using `:each` because it helps to ensure that the database is cleaned up each time. In this case, that's probably not necessary, though.